sktime/datatypes/_series/_check.py

```diff
@@ -0,0 +1,208 @@
+# -*- coding: utf-8 -*-
+"""Machine type checkers for Series scitype.
+
+Exports checkers for Series scitype:
+
+check_dict: dict indexed by pairs of str
+  1st element = mtype - str
+  2nd element = scitype - str
+elements are checker/validation functions for mtype
+
+Function signature of all elements
+check_dict[(mtype, scitype)]
+
+Parameters
+----------
+obj - object to check
+return_metadata - bool, optional, default=False
+    if False, returns only "valid" return
+    if True, returns all three return objects
+var_name: str, optional, default="obj" - name of input in error messages
+
+Returns
+-------
+valid: bool - whether obj is a valid object of mtype/scitype
+msg: str - error message if object is not valid, otherwise None
+        returned only if return_metadata is True
+metadata: dict - metadata about obj if valid, otherwise None
+        returned only if return_metadata is True
+    fields:
+        "is_univariate": bool, True iff series has one variable
+        "is_equally_spaced": bool, True iff series index is equally spaced
+        "is_empty": bool, True iff series has no variables or no instances
+"""
+
+__author__ = ["fkiraly"]
+
+__all__ = ["check_dict"]
+
+import numpy as np
+import pandas as pd
+
+
+VALID_INDEX_TYPES = (pd.Int64Index, pd.RangeIndex, pd.PeriodIndex, pd.DatetimeIndex)
+
+# whether the checks insist on freq attribute is set
+FREQ_SET_CHECK = False
+
+
+check_dict = dict()
+
+
+def check_pdDataFrame_Series(obj, return_metadata=False, var_name="obj"):
+
```

Comment:
missing docstring

Reply:
same for the other functions in this file

Reply:
these are private (so no docstring needed by linter), also all functions have the same signature which is detailed in the file docstring to avoid multiplication